bigframes/operations/semantics.py

```diff
@@ -61,30 +159,20 @@ def filter(self, instruction: str, model):
                 raise ValueError(f"Column {column} not found.")
 
         # Replace column references with names.
-        instruction = instruction.format(**{col: col for col in columns})
+        user_instruction = user_instruction.format(**{col: col for col in columns})
 
-        prompt_df = self._df.copy()
+        prompt_df = self._df[columns].copy()
```

Comment:
When `columns` specifies only a single column, let's verify its data type. If it's a single string (rather than a list),  `_df[columns]` might return a Pandas Series instead of a DataFrame. To ensure this behavior is correct, could we please a test case to double-check this scenario? Thanks!

Reply:
Sure. Test added.